_rules/form-field-accessible-name-is-descriptive-e0ca31.md

```diff
@@ -14,34 +14,42 @@ input_aspects:
+- has one of the following [semantic roles][semantic role]: `checkbox`, `combobox` (`select` elements), `listbox`, `menuitemcheckbox`, `menuitemradio`, `radio`, `searchbox`, `slider`, `spinbutton`, `switch` and `textbox`; and
+- is [visible][] or [included in the accessibility tree][]; and
+- has an [accessible name][].
 
-**Note**: The list of form field roles is derived by taking all the [ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1/) roles that:
+**Note**: The list of applicable [semantic roles][semantic role] is derived by taking all the [ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1/) roles that:
 
-- have a [semantic role][] that inherits from the [abstract](https://www.w3.org/TR/wai-aria/#abstract_roles) `input` or `select` role, and
-- does not have a [required context](https://www.w3.org/TR/wai-aria/#scope) role that itself inherits from one of those roles.
-- The `option` role is not part of the list of applicable roles, because it does not meet the definition of a [User interface component](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components). This means [WCAG 2.1](https://www.w3.org/TR/WCAG21/) does not require it to have an [accessible name](#accessible-name).
+- inherit from the [abstract](https://www.w3.org/TR/wai-aria/#abstract_roles) `input` or `select` role, and
+- do not have a [required context](https://www.w3.org/TR/wai-aria/#scope) role that itself inherits from one of those roles.
 
-**Note**: This rule is a partial check for WCAG 2.1 success criterion 2.4.6, which applies to all labels. "Label" is used in its general sense and includes text or other components with a text alternative that is presented to a user to identify a component within Web content.
+**Note:** The `option` role is not part of the list of applicable roles, because it has a required context role that inherits from the `select` role. Furthermore, `option` does not meet the definition of a [User interface component](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components). This means [WCAG 2.1](https://www.w3.org/TR/WCAG21/) does not require it to have an [accessible name][].
+
+**Note**: "Label" in WCAG is used in its general sense and includes text or other components with a text alternative that is presented to a user to identify a component within web content. That is, "label" in WCAG is not restricted to the `label` element of HTML or SVG.
 
 ## Expectation
 
-Each target element describes the purpose of the associated form field element.
+The [accessible name][] describes the purpose of the associated form field element. Both the [visual context][] of the element, and the [programmatically determined context][] which is [included in the accessibility tree][] can be used to differentiate the purpose from other form fields on the same page.
```

Comment:
This part "The [accessible name][] describes the purpose of the associated form field element." needs to be rewritten. Originally the rule targetted elements that were referenced by aria-labelledby so it made sense to talk about "associated form field elements". But now, we're addressing the accessible name of the form field element, so it becomes strange to talk about associated form field element. Suggestion: "The target element's [accessible name][] describes its purpose." or "The form field's [accessible name][] describes its purpose."

Remove "which is" from the second sentence.

Reply:
> Remove "which is" from the second sentence.

The intention of that sentence ("the programmatically determined context which is included in the accessibility tree") was to mean "the part of the programmatically determined context which is also included in the accessibility tree", thinking that part of the programmatically determined context that would be somehow left out of the accessibility tree is not really useful for explaining accessible name.

Do you mean that:
1. programmatically determined context is always included in the accessibility tree?
2. even the programmatically determined context which is **not** in the accessibility tree should be considered for this rule?
3. I should explicitly write "the **part of the** programmatically determined context which is included in the accessibility tree"?

Reply:
I got the meaning and if I had to select one of the options would be 3. But if you write "the programmatically determined context included in the accessibility tree" wouldn't it mean the same and be easier to read?

Reply:
Perfect!
I stupidly read `remove "which"` instead of `remove "which is"` and couldn't make sense out of it 🤦‍♂